src/s3.ts

```diff
@@ -141,19 +142,25 @@ export function grantAccessViaResourcePolicy(scope: IConstruct, id: string, prop
       conditions: {
```

Comment:
Can you wrap this in an opt-in flag as well? That's actually more the issue. Right now that policy insist the `x-amz-server-side-encryption` cannot be null. However when the default on the bucket is already KMS encryption its okay for the header to be null

Reply:
oh yes.  you're correct.  fix incoming.